### PR TITLE
Remove help_url,rename summary to symptom, and user_actions to diagnosis 

### LINKED ISSUE #88474: [Health API] Rename summary, user_actions, split messages and drop indicartor level help_url
### Description

This proposes a few field renamings and dropping the **indicator level** `help_url` field.

The renamings are : 
- `summary` to `symptom`.
- `user_actions` to `diagnosis`
- splitting the `user_action.messages` field into `cause` and `action`

Also, the `help_url` at the indicator level should be removed (as all indicators report a help_url via the user actions now)

### PR DESCRIPTION
**Pull request by andreidan**
This refactors the structure of the health API by renaming `summary` to `symptom`, 
renaming `user_actions` to `diagnosis`, splitting the `user_action.message` into two fields 
`diagnosis.cause` and `diagnosis.action`, and finally, removes the indicator level
`help_url` field (as the troubleshooting guides are offered through `diagnosis`)

This purposefully makes the `help_url` mandatory in the diagnosis as the indicator level
generic help is gone. 

Developers of new `diagnosis` will always have to provide a troubleshooting guide. 
There's some flexibility to help with splitting work by creating the URL alias that points to a generic 
troubleshooting guide when working on a new diagnosis - immediately followed by a PR that creates
a targetted troubleshooting guide for the new diagnosis. When the specific guide is completed it can 
be referenced under the already created URL alias.

Resolves #88474

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/health/HealthIndicatorResult.java [hunk 1]
```diff
@@ -18,30 +18,26 @@ public record HealthIndicatorResult(
...
-            builder.field("help_url", helpURL);
-        }
+        builder.field("symptom", symptom);
         if (details != null && HealthIndicatorDetails.EMPTY.equals(details) == false) {
             builder.field("details", details, params);
         }
         if (impacts != null && impacts.isEmpty() == false) {
             builder.field("impacts", impacts);
         }
         // TODO 83303: Add detail / documentation
```
Comment (jbaiera) [inline server/src/main/java/org/elasticsearch/health/HealthIndicatorResult.java:38]: Small nit while you're here: Might as well remove the todo now

**Review by jbaiera — APPROVED** [review]
LGTM! Thanks for going through all this. I had only minor nits.

**Comment by gmarouli** [comment]
This will be merged along with #88663 to make sure all API changes will be in the same release.

**Comment by andreidan** [comment]
@elasticmachine update branch

**Comment by andreidan** [comment]
@elasticmachine update branch

**Pull request merged**